mutex.go

```diff
@@ -0,0 +1,108 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package mutex
+
+import (
+	"regexp"
+	"time"
+
+	"github.com/juju/errors"
+)
+
+var (
+	validName = regexp.MustCompile("^[a-zA-Z][a-zA-Z0-9-]*$")
+)
+
+// Releaser defines the Release method that is the only thing that can be done
+// to a acquired mutex.
+type Releaser interface {
+	// Release makes the mutex available for others and invalidates the instance.
+	// Subsequent calls to Release will error.
+	Release() error
```

Comment:
I disagree. If you cannot release a lock you held, then by definition things are pretty fucked up. You cannot release the lock, and worse, no other process on the machine can aquire the lock.

This is such an extreme condition I think

a. we cannot allow the caller to ignore the error, and in fact I don't think there is a way they can handle it safely. Anything that quietly tucks it into a log file and goes through the internal worker restart process will not fix the sitaution.

b. All our locks are tied to the process, the invariant is, no matter what, killing the process will release the lock. So, if the lock cannot be released (I don't even know how this would happen), then by definition the way to release the lock is to kill the process via panic, or even os.Exit(1) (to make sure the panic is not caught)